zcash_keys/src/keys.rs

```diff
@@ -619,57 +644,62 @@ impl UnifiedFullViewingKey {
         let mut sapling = None;
         #[cfg(feature = "transparent-inputs")]
         let mut transparent = None;
+        let mut unknown_data = vec![];
+        let mut expiry_height = None;
+        let mut expiry_time = None;
+        let mut unknown_metadata = vec![];
```

Comment:
It's unclear whether `unified::Ufvk::decode(encoding)` enforces the requirement on the allowed receivers and metadata.

Reply:
The decoding of Unified containers currently disallows transparent-only UAs; this needs to change but will depend upon the resolution of https://github.com/zcash/librustzcash/pull/1135#discussion_r1515444301.

Reply:
Addressed in 673e1a21d28209103dce50ede358491e5978d007